tangram_bundler/__init__.py

```diff
@@ -1,35 +1,62 @@
 #!/usr/bin/env python
 
-import os, sys, glob, yaml, zipfile
+import os, sys, glob, yaml, zipfile, operator
 from urlparse import urljoin
+from functools import reduce
+
+def appendTexturePathCheckingGlobal(fileList, textureString, basePath, globalsNode):
+    if ("global." in textureString):
+        globalString = textureString.split('/')[-1]
+        globalPath = globalString.split('.', 1)[1].split('.')
+        globalValue = reduce(operator.getitem, globalPath, globalsNode)
+        fileList.append(os.path.relpath(globalValue, basePath))
+    else:
+        fileList.append(os.path.relpath(textureString, basePath))
 
 # Append uniform textures
 # - uniforms could reference a texture already loaded from textures: {} or could explicitly define a texture file
 
-def appendUniformTexturePath(fileList, basePath, rootNode, uniformTextureStr):
+def appendUniformTexturePath(fileList, basePath, rootNode, uniformTextureStr, globalsNode):
     referenceTexturePath = ""
     explicitUniformTexturePath = ""
     if ('textures' in rootNode) and (uniformTextureStr in rootNode['textures']):
         referenceTexturePath = os.path.relpath(rootNode['textures'][ uniformTextureStr]['url'], basePath)
     explicitUniformTexturePath = os.path.relpath(uniformTextureStr, basePath)
 
     if (os.path.exists(explicitUniformTexturePath)):
-        fileList.append(explicitUniformTexturePath)
+        appendTexturePathCheckingGlobal(fileList, explicitUniformTexturePath, basePath, globalsNode)
     elif (os.path.exists(referenceTexturePath)):
-        fileList.append(referenceTexturePath)
+        appendTexturePathCheckingGlobal(fileList, referenceTexturePath, basePath, globalsNode)
 
-def addUniformTextureDependency(fileList, basePath, rootNode, styleName, uniformName):
+def addUniformTextureDependency(fileList, basePath, rootNode, styleName, uniformName, globalsNode):
     key = rootNode['styles'][styleName]['shaders']['uniforms'][uniformName]
 
     if isinstance(key, basestring):
-        appendUniformTexturePath(fileList, basePath, rootNode, key)
+        appendUniformTexturePath(fileList, basePath, rootNode, key, globalsNode)
     elif isinstance(key, list):
         for textureStr in key:
             if isinstance(textureStr, basestring):
-                appendUniformTexturePath(fileList, basePath, rootNode, textureStr)
+                appendUniformTexturePath(fileList, basePath, rootNode, textureStr, globalsNode)
+
+def appendDrawRuleTexture(fileList, drawRule, basePath, globalsNode):
+    if 'texture' in drawRule:
+        appendTexturePathCheckingGlobal(fileList, drawRule['texture'], basePath, globalsNode)
+
+def appendLayerDrawRuleTextures(fileList, layerNode, basePath, globalsNode):
+    for key in layerNode:
+        if key == 'data' or key == 'filter' or key == 'visible' or key == 'enabled':
```

Comment:
Might want to move these to a list in more of a "config" section where they will be less buried and easier to spot if/when new reserved words are added.